lib/response.ex

```diff
@@ -71,6 +49,13 @@ defmodule HTTPServer.Response do
   end
 
   defp format_response_headers(headers) do
-    for {key, val} <- headers, into: "", do: "#{key}: #{val}#{@carriage_return}"
+    for {key, val} <- headers, into: "", do: "#{dasherize(key)}: #{val}#{@carriage_return}"
+  end
+
+  defp dasherize(key) do
+    to_string(key)
+    |> String.split("_")
+    |> Enum.map(&String.capitalize/1)
+    |> Enum.join("-")
   end
```

Comment:
I like this function, but I think the name could be improved. In this case what is technically a key, is actually a Response Header Name, and the implementation should probably match the domain language instead.

```suggestion
  defp format_header_name(header_name) do
    to_string(header_name)
    |> String.split("_")
    |> Enum.map(&String.capitalize/1)
    |> Enum.join("-")
  end
```